File: backend/src/main/java/si/feri/fitlink/checkin/CheckIn.java
package si.feri.fitlink.checkin;

import java.time.Instant;
import java.util.List;

import org.springframework.data.annotation.Id;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CheckIn {
    @Id
    private String id;
    private String traineeId;
    private Instant start;
    private double weightKg;
    private String photoUrl;
    private List<String> photoUrls;
    private String note;
    private TrainerComment trainerComment;
    private int overallEnergyLevel;
    private Instant createdAt;
    private boolean deletedByUser;

    @Data
    public static class TrainerComment {
        private String text;
        private Instant createdAt;
    }
}

File: mobile/src/types/checkin.ts
export interface CheckIn {
  id: string;
  traineeId: string;
  start: string;
  weightKg?: number;
  photoUrl?: string;
  photoUrls?: string[];
  note?: string;
  trainerComment?: CheckInComment;
  overallEnergyLevel: number;
  createdAt: string; 
}
export interface CheckInComment {
  text: string;
  createdAt: string;
}

export interface CheckInPhotoInput {
  uri: string;
  name?: string;
  type?: string;
}

export interface CheckInData {
  start: string;
  weightKg: number;
  photoUrl?: string;
  photoUrls?: string[];
  photo?: CheckInPhotoInput;
  photos?: CheckInPhotoInput[];
  note?: string;
  trainerComment?: CheckInComment;
  overallEnergyLevel: number;
}